feat!: Quicktime Movie Submission Workflows

Ahuge opened the pull request:
### What was the problem/requirement? (What/Why)
Current submissions were only supported for Frame based workflows and if you were rendering out a movie file, it would continuously overwrite it. See #73 
### What was the solution? (How)
Implement an automatic movie workflow if we detect the output of a comp is a movie extension
### What is the impact of this change?
This will automatically submit a single task job for the movie render if it detects it.
### How was this change tested?
Ran the workflow through all three submission queue types and tested both with a scene containing a frame comp and and a movie comp as well as just two movie comps.
### Did you run the "Job Bundle Output Tests"? If not, why not? If so, paste the test results here.

### Was this change documented?
No
### Is this a breaking change?
Yes, functionality for movie workflows has been changed
----

** This is built on top of #77 and should be on hold until that gets in


*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

Ahuge committed:
Update AfterEffects environment variable naming

This now matches the other new DCC naming scheme

Signed-off-by: Alex Hughes <ahughesalex@gmail.com>

Ahuge committed:
Implement Frame Chunking logic

This happens via a FrameStart, FrameEnd, ChunkSize and 2 internal
parameters that are just calculations based on the previous ones.

We've also reorganized some of the code to hopefully isolate it against
future changes.

Signed-off-by: Alex Hughes <ahughesalex@gmail.com>

Ahuge committed:
Save FramesPerTask as a setting

Signed-off-by: Alex Hughes <ahughesalex@gmail.com>

Ahuge committed:
Error handling around the batch file

Signed-off-by: Alex Hughes <ahughesalex@gmail.com>

Ahuge committed:
Start to implement a movie workflow

Signed-off-by: Alex Hughes <ahughesalex@gmail.com>

Ahuge committed:
[Fonts] Generate local font references

Signed-off-by: Alex Hughes <ahughesalex@gmail.com>

eherozhao commented:
Thanks for the PR! We are having a new dockable submitter with aerender and I think it should automatically solve this issue that old submitter had. Do you want to try it out and see if the new submitter works for this .mov output out of box?

yuanmich2 committed:
added a checkbox that adds an OpenJD step to convert image sequences to an uncompressed mov with ffmpeg

Ahuge committed:
Remove clientipc force include

The client IPC workflow is removed

Ahuge committed:
Automatically replace the FFMPEG fps with the comp's fps

Ahuge committed:
Insert _FPS_ token into mov step template

This will allow us to string replace it with the actual frame rate from the comp that we are submitting

Ahuge committed:
Merge pull request #4 from ArchPlatform/my/feature/ffmpeg-convert-to-mov

feature: added a checkbox that adds an OpenJD step to convert image sequences with ffmpeg
